sdk/sha256-hasher/src/lib.rs

```diff
@@ -28,7 +28,7 @@ impl Hasher {
 }
 
 #[cfg(target_os = "solana")]
-define_syscall!(fn sol_sha256(vals: *const u8, val_len: u64, hash_result: *mut u8) -> u64);
+pub use solana_define_syscall::definitions::sol_sha256;
```

Comment:
Looks like the CI failures are legit:

```
error: unused import: `solana_define_syscall::define_syscall`
 --> src/lib.rs:5:5
  |
5 | use solana_define_syscall::define_syscall;
```

From https://buildkite.com/anza/agave/builds/13734#019321ab-193e-4a6b-86b9-770af22325b7/6-6001

Since we don't lint with `target_os = solana`, it gets missed in other steps, but the import for `define_syscall` needs to be removed here. I think it's the only issue